File: LAB1/Insert/Insert.cpp
#include <iostream>
#include <time.h>
using namespace std;

void insertion(int arr[],   int tam)
{
    int aux, pos;
    for (int i=0;i<tam;i++)
    {
        pos=i;
        aux=arr[i];
        while((pos>0)&&(arr[pos-1]>aux))
        {
            arr[pos]=arr[pos-1];
            pos--;
        }
        arr[pos]=aux;
    }
    
}

int main(){

    int input,size;
    cin >> size;
    int array[size];
    for(int i = 0 ; i < size ; i++ )
    {
        cin >> input;
        array[i] = input;
    }
    unsigned t0,t1;
    t0 = clock();
    insertion(array, size);
    t1 = clock();
    double time = (double(t1-t0)/CLOCKS_PER_SEC);
    cout << time ;
    return 0;
    
}

File: LAB1/Insert/Insert.py
import random
import sys
from time import time

def insertSort(arr, tam):
   
    for i in range (tam):
        pos = i
        aux = arr[i]
        while(pos> 0) and (arr[pos-1] > aux):
            arr[pos] = arr[pos-1]
            pos = pos -1
        arr[pos] = aux

archivo = sys.stdin.readlines()
n = int(archivo[0])
arr = []
dato = archivo[1].split()
for i in range (n):
    arr.append(int(dato[i]))

start = time()
insertSort(arr, len(arr))
finish = time()
print(finish-start,end = '')
